### ISSUE TITLE
missing typ invariant for Ghost<...> field

### ISSUE THREAD

**Issue by tjhance** [verus-lang/verus#604]
This should pass, but it fails:

```
pub struct PageId {
    pub i: nat,
}

struct PageIdContainer {
    page_id: Ghost<PageId>,
}

spec fn a(page_id: PageId) -> bool;
spec fn b(page_id: PageId) -> bool;

proof fn test(pic: PageIdContainer) {
    let page_id = pic.page_id@;

    assume(a(page_id));
    assume(forall |page_id| #[trigger] a(page_id) ==> b(page_id));
    assert(b(page_id));  // fails
}
```

The reason seems to be that `page_id` never gets its type invariant assumed. If you add:

```
(assume (has_type (Poly%x!PageId. page_id~10@) TYPE%x!PageId.))
```

to the AIR file, then it passes.

[x.air.txt](https://github.com/verus-lang/verus/files/11737257/x.air.txt)
[x-fixed.air.txt](https://github.com/verus-lang/verus/files/11737258/x-fixed.air.txt)

**Comment by Chris-Hawblitzel** [comment]
Should be fixed by https://github.com/verus-lang/verus/commit/89172d6aebeda7bf445433c6803b79338b1f4bf2

**Comment by tjhance** [comment]
thanks for the quick fix! I added a regression test in eed97f1cb81cedede0d62e6ca8aacca66c7597b5
